### DISCUSSION TITLE
How would you extend filetypes while using LazyVim?

### DISCUSSION THREAD

**Discussion by surmish** [LazyVim/LazyVim#1088] [Q&A] [answered]
For getting [these](https://github.com/rafamadriz/friendly-snippets/blob/main/snippets/cpp/cppdoc.json) snippets in friendly-snippets plugin, in LuaSnip you need to extend file types as follows:

`require("luasnip").filetype_extend("cpp", { "cppdoc" })`

How would one go about writing this in `.config/nvim/lua/plugins/example.lua`? I don't want to copy the entire configuration for LuaSnip from LazyVim into my dotfiles. Simply add to it.

I tried adding the following to my `coding.lua` file but doesn't seem to trigger snippets for `cppdoc` filetype.

```lua
  {
    "L3MON4D3/LuaSnip",
    config = function()
      require("luasnip").filetype_extend("c", { "cdoc" })
      require("luasnip").filetype_extend("cpp", { "cppdoc" })
    end,
  },
```

**Comment by dpetka2001** [comment, marked as answer]
Add the following
```lua
  {
    "L3MON4D3/LuaSnip",
    dependencies = {
      "rafamadriz/friendly-snippets",
      config = function()
        require("luasnip").filetype_extend("c", { "cdoc" })
        require("luasnip.loaders.from_vscode").lazy_load()
      end,
    },
  },
```
The order does matter from what I've experimented with, `filetype_extend` should come before `lazy_load`. You could also do it in the Luasnip `config` function, but I decided to put it in the friendly-snipppets `config` function, because that's where LazyVim already calls the `lazy_load` by default, so I didn't want to also put it in the Luasnip `config` function and maybe have duplicate calls to `lazy_load` and thus having a possible impact on the loading time. So I decided to overwrite just the `config` function of friendly-snippets.
